### ISSUE TITLE
Disconnect clients when desynced too long

### ISSUE THREAD

**Issue by qreeves** [red-eclipse/base#673]
Currently while a client is desynced from the server in some way there's no indicator that things are failing other than the player not being able to pick up weapons or deal any damage. This can also cause a player to shoot projectiles only they can see. A client should probably be disconnected after a certain number of rejected messages, possibly with an auto-reconnect on their end to avoid the player being inconvenienced by the problem.

**Comment by acerspyro** [comment]
How do de-syncs happen to begin with? Is this just avoiding the problem rather than fixing it?

**Comment by qreeves** [comment]
I'd say it has to do with the architecture and the fact that RE has so many things going on at once in terms of network traffic back and forth due to the projectiles and all the other shit we've added over the years.

It isn't something we've been able to fix, but if it were something that could be fixed, it'd most likely be the fact the client assumes it's state is correct when the server made a different decision about said state. An auto reconnect might not even be necessary if we can just update the client's state to match that of the server.
